sdk/eventhubs/azure-eventhubs/src/main/java/com/azure/messaging/eventhubs/implementation/ReactorSender.java

```diff
@@ -211,13 +204,34 @@ public String getEntityPath() {
         return entityPath;
     }
 
+    @Override
+    public Mono<Integer> getLinkSize() {
+        if (this.hasConnected.get() && this.maxMessageSize > 0) {
+            return Mono.just(maxMessageSize);
+        }
+
+        return handler.getEndpointStates()
+            .takeUntil(state -> state == EndpointState.ACTIVE)
+            .timeout(timeout)
+            .then(Mono.fromCallable(() -> {
+                final UnsignedLong remoteMaxMessageSize = sender.getRemoteMaxMessageSize();
+
+                if (remoteMaxMessageSize != null) {
+                    this.maxMessageSize = remoteMaxMessageSize.intValue();
+                }
+
+                return this.maxMessageSize;
+            }));
+    }
+
     @Override
     public void close() {
         subscriptions.dispose();
         tokenManager.close();
         super.close();
     }
 
+
```

Comment:
...

Reply:
I feel compelled to add another blank line for you to find. 🤣